### ISSUE TITLE
`DIAGNOSTICS_PATH` silently defaults to current directory if esmvaltool cannot be imported

### ISSUE THREAD

**Issue by stefsmeets** [ESMValGroup/ESMValCore#826]
**Describe the bug**
I just had a problem with my esmvaltool installation, and I noticed that `DIAGNOSTICS_PATH` simply defaults to the CWD without any warning. In my opinion, it should either crash hard or make it very clear where it is looking for recipes. The current behaviour is not transparent, and it took me a while to figure it out.

**Comment by bouweandela** [comment]
Maybe a logger.info message displaying the diagnostics path would make this more obvious?

It would be nice if diagnostics_path was configurable in config user, with as default value the location of the esmvaltool installation or if that is not present, maybe the current working directory, but I'm not sure if that is a very sensible backup value.

**Comment by stefsmeets** [comment]
Agreed, that could be useful. I was thinking it would also make sense to configure it as an environment variable.

What do you think about further decoupling esmvaltool and esmvalcore? At the moment, esmvalcore presents itself as esmvaltool (just look at the entry point and `esmvalcore._main.py`) and installs the command line utility. This feels a bit the wrong way around. Right now esmvalcore depends on esmvaltool to ask where its diagnostics are located. When I think about it, esmvaltool should import esmvalcore and tell it where the diagnostics are located (easy to do if it is configurable). Then, you could have esmvalcore present a simple API to run recipes. Something to consider for the API.

**Comment by bouweandela** [comment]
Indeed the `esmvaltool` command is provided by the ESMValCore package. We kept the name because it's familiar, though if you have an idea for a better name, we could provide that as an alternative I guess and then move the `esmvaltool` command to the ESMValTool package at some point.

It is possible to use the ESMValCore package and the `esmvaltool` command without installing the ESMValTool package and this works just fine, but it is not well documented what files and directories are expected if you want things to be fully working. This is something that definitely could use some improvements.

**Comment by stefsmeets** [comment]
Yeah, I was thinking too that the esmvaltool command line program logically belongs to esmvaltool. We could have a trimmed down version for esmvalcore that just runs recipes and handle configs (pretty much what it currently is). We could import the base command line in ESMValTool and customize it with more esmvaltool related commands if necessary. I think this would be good to have better isolation between the two packages. ESMValTool is already installed as a python package anyway, so it should not be too difficult.
